sanctions/apps/api/v1/views.py

```diff
@@ -73,15 +74,63 @@ def post(self, request):
                 city,
                 country
             )
-            response = {'total': sdn_fallback_hit_count}
+            sdn_check_response = {'total': sdn_fallback_hit_count}
 
-        hit_count = response['total']
+        hit_count = sdn_check_response['total']
         if hit_count > 0:
             logger.info(
                 'SDNCheckView request received for lms user [%s]. It received %d hit(s).',
                 lms_user_id,
                 hit_count,
             )
+            # write record to our DB that we've had a positive hit, including
+            # any metadata provided in the payload
+            metadata = payload.get('metadata', {})
+            username = payload.get('username')
+            system_identifier = payload.get('system_identifier')
+            sanctions_type = 'ISN,SDN'
+            # This try/except is here to make us fault tolerant. Callers of this
+            # API should not be held up if we are having DB troubles. Log the error
+            # and continue through the code to reply to them.
+            try:
+                SanctionsCheckFailure.objects.create(
+                    full_name=full_name,
+                    username=username,
+                    lms_user_id=lms_user_id,
+                    city=city,
+                    country=country,
+                    sanctions_type=sanctions_type,
+                    system_identifier=system_identifier,
+                    metadata=metadata,
+                    sanctions_response=sdn_check_response,
+                )
+            except Exception as err:  # pylint: disable=broad-exception-caught
+                error_message = (
+                    'Encountered error creating SanctionsCheckFailure. %s '
+                    'Data dump follows to capture information on the hit: '
+                    'lms_user_id: %s '
+                    'username: %s '
+                    'full_name: %s '
+                    'city: %s '
+                    'country: %s '
+                    'sanctions_type: %s '
+                    'system_identifier: %s '
+                    'metadata: %s '
+                    'sanctions_response: %s '
+                )
+                logger.exception(
+                    error_message,
+                    err,
+                    lms_user_id,
+                    username,
+                    full_name,
+                    city,
+                    country,
+                    sanctions_type,
+                    system_identifier,
+                    metadata,
+                    sdn_check_response,
+                )
```

Comment:
We should create an alert when this happens that will create a ticket in our backlog or something similar

Reply:
👍 read my mind. i can create a splunk alert for this as part of this ticket